From d379cf1ddd8c0b97de9a9c7ba1710056c3d2d583 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 11 Nov 2010 15:50:42 +0800
Subject: [PATCH 093/502] ENGR00000001 Get max buffer size before set it.

It's not good to hardcode the buffer size in the source.
We can get the max buffer size by alsa lib api.

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 modules/alsa/alsa_imx5x.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/modules/alsa/alsa_imx5x.cpp b/modules/alsa/alsa_imx5x.cpp
index 05d6a26..1d75161 100755
--- a/modules/alsa/alsa_imx5x.cpp
+++ b/modules/alsa/alsa_imx5x.cpp
@@ -252,6 +252,13 @@ status_t setHardwareParams(alsa_handle_t *handle)
     else
         LOGV("Set sample rate to %u HZ", requestedRate);
 
+    // get the max buffer size we can set
+    err = snd_pcm_hw_params_get_buffer_size_max(hardwareParams, &bufferSize);
+    if (err < 0) {
+        LOGE("Unable to get max buffer size:  %s", snd_strerror(err));
+        goto done;
+    }
+
     // Make sure we have at least the size we originally wanted
     err = snd_pcm_hw_params_set_buffer_size(handle->handle, hardwareParams,
             bufferSize);
-- 
1.8.0

